From d16ce743d8d5a284ac95f054e6866e5c80850460 Mon Sep 17 00:00:00 2001
From: kszaq <kszaquitto@gmail.com>
Date: Wed, 29 May 2019 22:43:00 +0200
Subject: [PATCH] init: ondemand tuning

Set all ondemand tunables like it's done for RPi.
Thanks to these GUI animations feel smoother,
particularly on ARM targets.
---
 packages/sysutils/busybox/scripts/init | 17 +++++++++--------
 1 file changed, 9 insertions(+), 8 deletions(-)

diff --git a/packages/sysutils/busybox/scripts/init b/packages/sysutils/busybox/scripts/init
index 616d552d796..d91c5311d72 100755
--- a/packages/sysutils/busybox/scripts/init
+++ b/packages/sysutils/busybox/scripts/init
@@ -924,14 +924,15 @@ prepare_sysroot() {
 # hide kernel log messages on console
 echo '1 4 1 7' > /proc/sys/kernel/printk
 
-# set ondemand up_threshold
-if [ -e /sys/devices/system/cpu/cpufreq/ondemand/up_threshold ]; then
-  echo 50 > /sys/devices/system/cpu/cpufreq/ondemand/up_threshold
-else
-  for f in $(ls /sys/devices/system/cpu/cpufreq/policy*/ondemand/up_threshold 2>/dev/null) ; do
-    echo 50 > $f
-  done
-fi
+# Configure frequency scaling properties
+for policy in /sys/devices/system/cpu/cpufreq/ondemand /sys/devices/system/cpu/cpufreq/policy*/ondemand ; do
+  if [ -d $policy ] ; then
+    echo 1 > $policy/io_is_busy
+    echo 50 > $policy/up_threshold
+    echo 100000 > $policy/sampling_rate
+    echo 50 > $policy/sampling_down_factor
+  fi
+done
 
 # run platform_init script if exists
 if [ -f "./platform_init" ]; then
